proving_system/stark/tests/integration_tests.rs

```diff
@@ -248,6 +297,83 @@ fn test_prove_rap_fib() {
 
     let fibonacci_rap = FibonacciRAP::new(context);
 
-    let result = prove(&trace_cols, &fibonacci_rap);
-    assert!(verify(&result, &fibonacci_rap));
+    let result = prove(&trace_cols, &fibonacci_rap, &mut ());
+    assert!(verify(&result, &fibonacci_rap, &()));
+}
+
+#[test_log::test]
+fn test_prove_dummy() {
+    let trace_length = 16;
+    let trace = dummy_air::dummy_trace(trace_length);
+
+    let context = AirContext {
+        options: ProofOptions {
+            blowup_factor: 2,
+            fri_number_of_queries: 1,
+            coset_offset: 3,
+        },
+        trace_length,
+        trace_columns: 2,
+        transition_degrees: vec![2, 1],
+        transition_exemptions: vec![0, 2],
+        transition_offsets: vec![0, 1, 2],
+        num_transition_constraints: 2,
+    };
+
+    let dummy_air = dummy_air::DummyAIR::from(context);
+
+    let result = prove(&trace, &dummy_air, &mut ());
+    assert!(verify(&result, &dummy_air, &()));
+}
+
+#[test_log::test]
+fn test_prove_cairo_simple_program_malicious() {
+    let base_dir = env!("CARGO_MANIFEST_DIR");
+    let dir_trace = base_dir.to_owned() + "/src/cairo_vm/test_data/simple_program.trace";
+    let dir_memory = base_dir.to_owned() + "/src/cairo_vm/test_data/simple_program.mem";
+
+    let malicious_raw_trace = CairoTrace::from_file(&dir_trace).unwrap();
+    let malicious_memory = CairoMemory::from_file(&dir_memory).unwrap();
+
+    let proof_options = ProofOptions {
+        blowup_factor: 4,
+        fri_number_of_queries: 1,
+        coset_offset: 3,
+    };
+
+    let program_size = 5;
+    let mut malicious_program = vec![];
+    for i in 1..=program_size as u64 {
+        malicious_program.push(malicious_memory.get(&i).unwrap().clone());
+    }
+
+    let mut real_program = malicious_program.clone();
+    real_program[1] = FieldElement::from(5);
+    real_program[3] = FieldElement::from(5);
+
+    let cairo_air = cairo::CairoAIR::new(proof_options, 16, malicious_raw_trace.steps());
+
+    let first_step = &malicious_raw_trace.rows[0];
+    let last_step = &malicious_raw_trace.rows[malicious_raw_trace.steps() - 1];
+
+    let mut public_input = PublicInputs {
+        pc_init: FE::from(first_step.pc),
+        ap_init: FE::from(first_step.ap),
+        fp_init: FE::from(first_step.fp),
+        pc_final: FE::from(last_step.pc),
+        ap_final: FE::from(last_step.ap),
+        program: malicious_program,
+        range_check_min: None,
+        range_check_max: None,
+        num_steps: malicious_raw_trace.steps(),
+    };
+
+    let result = prove(
+        &(malicious_raw_trace, malicious_memory),
+        &cairo_air,
+        &mut public_input,
+    );
+
+    public_input.program = real_program;
+    assert!(!verify(&result, &cairo_air, &public_input));
 }
```

Comment:
This test has a bad name and isn't documented. We already have a couple of malicious provers, it's not clear in this one what's malicious about it.